src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -0,0 +1,965 @@
+    {
+        return jsonFilePath;
+    }
+
+    json::JsonObject FancyZonesData::GetPersistFancyZonesJSON()
+    {
+        std::wstring save_file_path = GetPersistFancyZonesJSONPath();
+
+        auto result = json::from_file(save_file_path);
+        if (result)
+        {
+            return *result;
+        }
+        else
+        {
+            return json::JsonObject();
+        }
+    }
+
+    int FancyZonesData::GetAppLastZone(HWND window, PCWSTR appPath, _Out_ PINT iZoneIndex) const
+    {
+        *iZoneIndex = -1;
+
+        if (auto monitor = MonitorFromWindow(window, MONITOR_DEFAULTTONULL))
+        {
+            TAppPath path{ appPath };
+            if (appZoneHistoryMap.contains(path))
+            {
+                *iZoneIndex = appZoneHistoryMap.at(path).zoneIndex;
+            }
+        }
+        return *iZoneIndex;
+    }
+
+    // Pass -1 for the zoneIndex to delete the entry from the map
+    bool FancyZonesData::SetAppLastZone(HWND window, PCWSTR appPath, DWORD zoneIndex)
+    {
+        if (auto monitor = MonitorFromWindow(window, MONITOR_DEFAULTTONULL))
+        {
+            if (zoneIndex == -1)
+            {
+                appZoneHistoryMap.erase(TAppPath{ appPath });
+            }
+            else
+            {
+                //TODO(stefan) provide correct uuid in the future
+                appZoneHistoryMap[TAppPath{ appPath }] = AppZoneHistoryData{ L"", static_cast<int>(zoneIndex) };
+            }
+            return true;
+        }
+        return false;
+    }
+
+    void FancyZonesData::SetActiveZoneSet(const std::wstring& uniqueID, const std::wstring& uuid)
+    {
+        if (!uuid.empty() && deviceInfoMap.find(uniqueID) != deviceInfoMap.end())
+        {
+            deviceInfoMap[uniqueID].activeZoneSet.uuid = uuid;
+        }
+    }
+
+    void FancyZonesData::SetDeviceInfoToTmpFile(const DeviceInfoJSON& deviceInfo, const std::wstring& tmpFilePath) const
+    {
+        json::JsonObject deviceInfoJson = DeviceInfoJSON::ToJson(deviceInfo);
+        json::to_file(tmpFilePath, deviceInfoJson);
+    }
+
+    void FancyZonesData::GetDeviceInfoFromTmpFile(const std::wstring& uniqueID, const std::wstring& tmpFilePath)
+    {
+        if (!deviceInfoMap.contains(uniqueID))
+        {
+            // Creates entry in map when ZoneWindow is created
+            // TODO(stefan): provide valid uuid
+            deviceInfoMap[uniqueID] = DeviceInfoData{ ZoneSetData{ L"", ZoneSetLayoutType::Grid, 1 }, true, 16, 3 };
+        }
+
+        if (std::filesystem::exists(tmpFilePath))
+        {
+            auto zoneSetJson = json::from_file(tmpFilePath);
+            if (zoneSetJson.has_value())
+            {
+                const auto deviceInfo = DeviceInfoJSON::FromJson(*zoneSetJson);
+                if (deviceInfo.has_value())
+                {
+                    deviceInfoMap[uniqueID] = deviceInfo->data;
+                }
+            }
+
+            DeleteFileW(tmpFilePath.c_str());
```

Comment:
It's unexpected for a function named `GetDeviceInfoFromTmpFile` to also remove its input file argument.